lib/Frontend/Frontend.cpp

```diff
@@ -884,6 +884,79 @@ void CompilerInstance::performParseOnly(bool EvaluateConditionals) {
          "Loaded a module during parse-only");
 }
 
+void CompilerInstance::performParseAndNameBindingOnly() {
```

Comment:
Can this share code with any of the other actions?

Reply:
I can probably fold it into `performSema` now, but we may not necessarily be calling `performNameBinding` in the future for this path.

Reply:
Yeah, something about this feels a bit wrong to me too: the extent to which `performParseOnly` and the parsing-part of `parseAndCheckTypes` have already diverged seems regrettable, and this is just going to keep digging that hole by further duplicating and diverging. Can we refactor these paths without making them awful?

Reply:
I'm emphatically with Graydon and Slava here. It already really bothers me that there were two main routines, `performParseOnly` and `performSema`. You would be doing the code base a great service if you could figure out how to factor these together so it were crystal clear which operations went with which alternative and which ones were common.

Reply:
I think `performParseOnly` and `performSema` actually *are* significantly different, but `performSema` and this new operation are not. Please do make this a mode for `performSema` instead.